scripts/migrate-test/main.go

```diff
@@ -0,0 +1,148 @@
+package main
+
+import (
+	"archive/zip"
+	"bytes"
+	"database/sql"
+	"flag"
+	"fmt"
+	"io/fs"
+	"os"
+	"os/exec"
+	"regexp"
+
+	"github.com/google/go-cmp/cmp"
+	"golang.org/x/xerrors"
+
+	"github.com/coder/coder/v2/coderd/database/dbtestutil"
+	"github.com/coder/coder/v2/coderd/database/migrations"
+)
+
+// This script validates the migration path
+// between two versions.
+// It performs the following actions:
+// Given OLD_VERSION and NEW_VERSION:
+//  1. Checks out $OLD_VERSION and inits schema at that version.
+//  2. Checks out $NEW_VERSION and runs migrations.
+//  3. Compares database schema post-migrate to that in VCS.
+//     If any diffs are found, exits with an error.
+func main() {
```

Comment:
Who/what/when is going to run this script? The hidden question is: can somebody forget to run it and there will be a disaster?

Reply:
Point 👍 I need to add this to the CI workflow so it gets run on each commit.